lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -598,6 +598,12 @@ def safely_acquire_lock_for_table(table, mode: :access_exclusive, &block)
     Thread.current[__method__] = nil unless nested_target_table
   end
 
+  # since rails versions < 7.1 has bug which does not handle symbol for
+  # constraint name, we are converting name to string explicitly to solve that.
+  def unsafe_remove_check_constraint(table, name:, **options)
```

Comment:
I think after https://github.com/braintree/pg_ha_migrations/pull/98 we might consider making this an override of `raw_remove_check_constraint` instead of `unsafe_*` since it's effectively back-porting a bug fix as opposed to providing HA safety for DDL.

@rkrage Curious your thoughts.

Reply:
Definitely agree with overriding `raw_` instead. updated in this commit: https://github.com/braintree/pg_ha_migrations/pull/102/commits/e66dea125bca3668daa81b2991733211766a467f, but if others disagree, we can revert.

Also, not sure if the method & test belong in the `safe_` files they are currently in? The `safe_statements.rb` file has some `unsafe_` method definitions already, so I left it as is for now.

Reply:
Also agree that we should be overriding the raw method, but I think maybe it should go in `unsafe_statements.rb`. @jcoleman is there a reason we sometimes have unsafe methods in the safe statements file? I generally put safe and unsafe variants of _custom_ methods in the safe file, just so the code is close together. But for basic wrappers of existing Rails methods... maybe the unsafe file is more correct.

Reply:
I think it might have something to do with the module prepending order, but without testing it I don't remember details. Possible there's a commit message explaining it somewhere.

Reply:
Okay, then I think this is fine for now. I might try to follow up with a PR to move all these methods into more "correct" files and we'll see what tests break.

Reply:
Oh, I think `unsafe_rename_enum_value` is there simply because there isn't an analogue in ActiveRecord.

But the other ones -- like `unsafe_add_column` I think are there because they reference `safely_acquire_lock_for_table` which is in that same file (and it'd be weird for a module higher up there ancestor chain to call a method at a lower level). We could move that function though.

Reply:
Maybe just move the utility methods to separate module that gets prepended higher up the chain then?

Reply:
Yeah, we could probably do that. Separate work/PR though?